### PR TITLE
Minor refactor of tests

### PR DESCRIPTION
**Pull request by tschuy**

### CODE REVIEW

File: tests/test.js [hunk 1]
```diff
@@ -12,7 +12,35 @@ var knex = require('knex')(knexfile.mocha);
...
+                        knex('projectslugs').del().then(function() {
+                            sqlFixtures.destroy().then(function() {
+                                done();
+                            });
+                        });
+                    });
+                });
+            });
+        });
+    });
```
Comment (iankronquist) [inline tests/test.js]: Is there a better way to do this? This is the definition of callback hell.

Comment (iankronquist) [inline tests/test.js]: https://github.com/petkaantonov/bluebird/blob/master/API.md#all---promise
I think this is what we want.

Comment (tschuy) [inline tests/test.js]: I don't think so. sqlite doesn't do multiple deletes at once, and a raw knex appeared to pass, but then gave a SQLITE_CONSTRAINT: UNIQUE error.

**Comment by tschuy** [comment]
https://pastebin.osuosl.org/29976/

**Comment by iankronquist** [comment]
For the future, these requests need to be made serially, not in parallel because sqlite isn't a real database. The Promise.all approach does not work, and chaining 'then' calls is just as ugly as what's there right now.

**Comment by ElijahCaine** [comment]
That's very weird that the tests failed. They passed last night before I pushed. I'll revert the last commit and we can move on.

**Comment by iankronquist** [comment]
@ElijahCaine I think we need to specify node v0.12 in the .travis.yml.

**Comment by ElijahCaine** [comment]
@iankronquist 
TravisCI
```
$ nvm install 0.10

######################################################################## 100.0%

Now using node v0.10.40

$ node --version

v0.10.40
```
I think you're right :smile:

**Comment by tschuy** [comment]
I had been meaning to get around to fixing that. My bad!

**Comment by iankronquist** [comment]
@ElijahCaine any reason you aren't doing this?
```javascript
// databsePromises gets iterated over by Promise,
// this avoids the classic nested callback hell.
var clearDatabasePromises = [
knex('projects').del(),
knex('activities').del(),
knex('users').del(),
knex('times').del(),
knex('activityslugs').del(),
knex('projectslugs').del(),
sqlFixtures.destroy(),
];



```

**Comment by ElijahCaine** [comment]
@iankronquist Can you clarify your concerns about the changes I made? We can revert them, I want to make sure it's necessary to do so.

File: tests/test.js [hunk 2]
```diff
@@ -21,13 +52,7 @@ describe('Endpoints', function() {
...
-        // Clear SQLite indexes
-        knex.raw('delete from sqlite_sequence').then(function() {
-            sqlFixtures.create(knexfile.mocha, testData).then(function() {
-                done();
-            });
-        });
+        reloadFixtures(done);
     });
 
     afterEach(function(done) {
```
Comment (tschuy) [inline tests/test.js]: This should just be clearDatabase();

**Comment by tschuy** [comment]
Looks good to me. Get another review and let's get this thing in!

**Comment by iankronquist** [comment]
Looks good. Tests pass, linter, etc. Our coverage is dropping, but that's no fault of this pr.

**Pull request merged**